video/out/vo_gpu_next.c

```diff
@@ -763,8 +763,13 @@ static void update_options(struct vo *vo)
 {
     struct priv *p = vo->priv;
     pl_options pars = p->pars;
-    bool changed = m_config_cache_update(p->opts_cache);
-    changed = m_config_cache_update(p->next_opts_cache) || changed;
+
+    bool opts_changed = m_config_cache_update(p->opts_cache);
+    bool next_opts_changed = m_config_cache_update(p->next_opts_cache);
+    bool changed = opts_changed | next_opts_changed;
+
+    if (next_opts_changed)
+        p->want_reset = true;
```

Comment:
Do you actually need put reset here?

Reply:
It is empirically needed to fix the image-lut issue for me.

My understanding of the code is that `image_lut` is attached in `map_frame()`. So applying the updated option immediately requires remapping frames, and `want_reset` is the way to do it.

Is my understanding incorrect.

Reply:
Now that I actually look at it, the other lut options already update on pause. In fact, the manual entry for `--image-lut` implies to me that the picture should actually not change in this case: `Specifies  a  custom  LUT  file  (in  Adobe  .cube  format)  to  apply  to  the colors during image decoding.`. Keyphrase here being "during image decoding". You already decoded the image so it would not change. If you seek around while paused, then it applies which is expected.

Doing a reset seems awfully heavy handed to me since as far as I know, nothing else requires it.

Reply:
This feels like a *post hoc* understanding to me. From an end user perspective, can't the same argument be made about `MAIN` and pre-`MAIN` hooks in GLSL shaders? Those apply on pause, as an end user would expect.

> Doing a reset seems awfully heavy handed to me since as far as I know, **nothing else requires it**.

There is another option which is to check if `image-lut` or `image-lut-type` where specifically changed. But I myself wasn't sure about nothing else really requiring it, especially with `libplacebo-opts` being a part of the same struct. So my thought process was that a reset on any `next_opts` change was the simpler, safer, and more consistent approach. And it's not like `next_opts` field values often change willy-nilly.

----

Anyway, I like the update-on-pause behavior, so I will keep it locally. It's up to you whether this is a change for everyone or not.